.github/workflows/Tests.yml

```diff
@@ -398,24 +401,29 @@
       CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
     steps:
     - uses: actions/checkout@v4
-    - name: Setup Python ${{ matrix.python-version }}
+    - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies for python ${{ matrix.python-version }} on ${{ matrix.os }}
       run: |
-        pip install --upgrade pip setuptools wheel
-        pip install -r ./requirements.txt ;
-        pip install coverage ;
-        pip install pytest ;
-        pip install pytest-cov ;
-    - name: Install code-climate tools for ${{ matrix.python-version }}
-      if: ${{ runner.os }} != "Linux"
-      run: |
-        if [ $OS == macos-latest ] ; then curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-darwin-amd64 > ./cc-test-reporter || true ; fi ;
-        if [ $OS == macos-latest ] ; then chmod +x ./cc-test-reporter 2>/dev/null || true ; fi
-        if [ $OS == macos-latest ] ; then ./cc-test-reporter before-build || true ; fi
+        pip install --upgrade "pip>=21.0" "setuptools>=45.0" "wheel>=0.37" "build>=1.0.1";
+        pip install --upgrade -r ./requirements.txt ;
+        pip install --upgrade -r ./tests/requirements.txt || true ;
+        pip install --upgrade -r ./docs/requirements.txt || true ;
+    - name: Install Darwin code-climate tools for ${{ matrix.python-version }}
+      if: ${{ !cancelled() && runner.os != 'Linux' }}
+      run: |
+        if [ "$OS" == "macos-latest" ] ; then curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-darwin-amd64 > ./cc-test-reporter || true ; fi ;
+        if [ "$OS" == "macos-latest" ] ; then chmod +x ./cc-test-reporter 2>/dev/null || true ; fi
+        if [ "$OS" == "macos-latest" ] ; then ./cc-test-reporter before-build || true ; fi
       shell: bash
+    - name: Install Linux code-climate tools for ${{ matrix.python-version }}
+      if: ${{ !cancelled() && runner.os == 'Linux' }}
+      run: |
+        curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter || curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-arm64 > ./cc-test-reporter || true ;
+        chmod +x ./cc-test-reporter 2>/dev/null || true ;
+        ./cc-test-reporter before-build || true ;
```

Comment:
Interesting idea. Please open a new GH issue for this suggestion called "Testing out improved conditionals in GH actions" with this suggestion as the new issue's description. This will have to be tested in a separate PR/MR.